syntax = "proto2";

// protoc 3.4.0

package pbi;

message UnsignedPbiData {
  message Data {
    required EncryptedBlock hashes = 1;
    repeated EncryptedBlock push = 2;
    repeated EncryptedBlock pull = 3;
  }
  required Data data = 1;
  required bytes trailer = 2;
}

message EncryptedBlock {
    required bytes data = 1;
}

message FileHashes {
    repeated HashedFile files = 1;
    repeated bytes others = 2;
}

message HashedFile {
    required bytes path = 1;
    required bytes hash = 2;
}

message PushBindings {
    repeated Binding bindings = 1;
}

message PullBindings {
    repeated Binding bindings = 1;
}

message Binding {
    required bytes key = 1;
    required Value value = 2;
}

message Value {
    oneof value {
        bool boolean = 1;
        int32 integer = 5;
        bytes string = 9;
    }
}
